cli.go

```diff
@@ -0,0 +1,502 @@
+package jwtauth
+
+import (
+	"errors"
+	"fmt"
+	"net/http"
+	"os"
+	"os/exec"
+	"os/signal"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/hashicorp/vault/api"
+)
+
+const defaultMount = "oidc"
+const defaultPort = "8300"
+
+type CLIHandler struct{}
+
+type loginResp struct {
+	secret *api.Secret
+	err    error
+}
+
+func (h *CLIHandler) Auth(c *api.Client, m map[string]string) (*api.Secret, error) {
+	// handle ctrl-c while waiting for the callback
+	ch := make(chan os.Signal, 1)
+	signal.Notify(ch, os.Interrupt)
+	defer signal.Stop(ch)
```

Comment:
Could ch be named something like ctrlC or interrupt or something more descriptive?

Reply:
Renamed to `sigintCh`